File: lib/models/Student.dart
import 'dart:convert';

class Student {
  final String id;
  final String fullName;
  final String location;
  final String email;
  final String phoneNumber;
  String? picturePath;
  final String institution;

  final String classStudies;

  Student({
    required this.id,
    required this.fullName,
    required this.location,
    required this.email,
    required this.phoneNumber,
    this.picturePath,
    required this.institution,
    required this.classStudies,
  });

  Map<String, dynamic> toMap() {
    return {
      'id': id,
      "fullName": fullName,
      "location": location,
      "email": email,
      "phoneNumber": phoneNumber,
      "picturePath": picturePath,
      "institution": institution,
      "classStudies": classStudies,
    };
  }

  String toJson() => json.encode(toMap());
}

File: server/models/Student.js
const mongoose = require("mongoose");

const Schema = mongoose.Schema;

const studentSchema = new Schema({
  fullName: {
    type: String,
    required: true,
    min: 2,
    max: 50,
  },
  email: {
    type: String,
    required: true,
    unique: true,
  },

  classStudies: {
    type: String,  //change it 
    required: true,
  },
  location: {
    type: String,
    required: true,
    max: 50,
  },
  phoneNumber: {
    type: String,
    required: true,
    min: 6,
    max: 12,
  },
  picturePath: {
    type: String,
    default: "",
    required: false,
  },
  institution: {
    type: String,
    default: ""
  },
  teacherId: {
    type: Array,
    //required: true,
    default: [],
  },
  advertisementId: {
    type: Array,
    default: [],
  },
});

module.exports = mongoose.model("Student", studentSchema);
